react/components/EXPERIMENTAL_Select/index.js

```diff
@@ -22,11 +22,22 @@ class Select extends Component {
   constructor(props) {
     super(props)
 
+    this.state = {
+      searchTerm: undefined,
+    }
+
     this.inputId = `react-select-input-${uuid()}`
   }
 
-  componentDidUpdate() {
-    document.getElementById(this.inputId).focus()
+  componentDidUpdate(prevProps, prevState) {
+    const { searchTerm } = this.state
+    const { searchTerm: prevSearchTerm } = prevState
+    const { loading } = this.props
+    const { loading: prevLoading } = prevProps
+
+    if (searchTerm !== prevSearchTerm || loading !== prevLoading) {
+      document.getElementById(this.inputId).focus()
```

Comment:
Use ref instead of `getElementById`.

Also, I think that's not a very reliable solution. 

Can you apply focus just when the user selects an option?

Reply:
As discussed personally, the component don't provide ref.

Also, the searchTerm is not passed as props, that's why its kept inside as a state.